src/emitter.rs

```diff
@@ -334,3 +385,29 @@ fn need_quotes(string: &str) -> bool {
         || string.parse::<i64>().is_ok()
         || string.parse::<f64>().is_ok()
 }
+
+/// Check if the string can be expressed a valid literal block scalar.
+/// The YAML spec supports all of the following in block literals except #xFEFF:
+///      #x9 | #xA | [#x20-#x7E]          /* 8 bit */
+///   | #x85 | [#xA0-#xD7FF] | [#xE000-#xFFFD] /* 16 bit */
+///   | [#x10000-#x10FFFF]                     /* 32 bit */
+fn is_valid_literal_block_scalar(string: &str) -> bool {
+    string.chars().all(|character: char|
+        matches!(character, '\t' | '\n' | '\x20'..='\x7e' | '\u{0085}' | '\u{00a0}'..='\u{d7fff}'))
+}
+
+#[cfg(test)]
+mod test {
+    use super::YamlEmitter;
+    use crate::YamlLoader;
+
+    #[test]
+    fn test_multiline_string() {
+        let input = r#"{foo: "bar!\nbar!", baz: 42}"#;
+        let parsed = YamlLoader::load_from_str(input).unwrap();
+        let mut output = String::new();
+        let mut emitter = YamlEmitter::new(&mut output);
+        emitter.multiline_strings(true);
+        emitter.dump(&parsed[0]).unwrap();
```

Comment:
This needs an assert, which might be this (I'm unsure)

```suggestion
        emitter.dump(&parsed[0]).unwrap();
        assert_eq!(output.as_str(), "\
---
foo: |
  bar!
  bar!
baz: 42");
```

Reply:
Yup, the `assert_eq!` here should be handling the assert for this (doc)test. I noticed `/// # {` and `/// # }` comments in the docstring example. I went ahead and removed those since they're not necessary.